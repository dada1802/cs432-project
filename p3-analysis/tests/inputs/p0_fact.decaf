def int fact(int n)
{
    if (n == 1)
    {
        return 1;
    }

    return n * fact(n - 1);
}

def int main()
{
    int a;
    a = fact(4);
    return a;
}