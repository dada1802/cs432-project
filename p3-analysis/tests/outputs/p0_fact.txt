Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 fact1 : (int) -> int
 fact2 : (int) -> int
 fact3 : (int) -> int
 main : () -> int

  FuncDecl name="fact1" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:

        Block [line 3]
        SYM TABLE:

  FuncDecl name="fact2" return_type=int parameters={n:int} [line 9]
  SYM TABLE:
   n : int

    Block [line 10]
    SYM TABLE:
     result : int

        Block [line 14]
        SYM TABLE:

  FuncDecl name="fact3" return_type=int parameters={n:int} [line 21]
  SYM TABLE:
   n : int

    Block [line 22]
    SYM TABLE:
     sum : int

        Block [line 26]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 33]
  SYM TABLE:

    Block [line 34]
    SYM TABLE:

