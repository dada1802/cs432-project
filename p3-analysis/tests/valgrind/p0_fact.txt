==3079975== Memcheck, a memory error detector
==3079975== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3079975== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3079975== Command: ../decaf inputs/p0_fact.decaf
==3079975== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 fact : (int) -> int
 main : () -> int

  FuncDecl name="fact" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:

        Block [line 4]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 11]
  SYM TABLE:

    Block [line 12]
    SYM TABLE:
     a : int

==3079975== 
==3079975== HEAP SUMMARY:
==3079975==     in use at exit: 0 bytes in 0 blocks
==3079975==   total heap usage: 2,020 allocs, 2,020 frees, 238,800 bytes allocated
==3079975== 
==3079975== All heap blocks were freed -- no leaks are possible
==3079975== 
==3079975== For lists of detected and suppressed errors, rerun with: -s
==3079975== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
