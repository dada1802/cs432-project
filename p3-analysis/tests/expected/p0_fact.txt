Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 fact : (int) -> int
 main : () -> int

  FuncDecl name="fact" return_type=int parameters={n:int} [line 1]
  SYM TABLE:
   n : int

    Block [line 2]
    SYM TABLE:

        Block [line 4]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 11]
  SYM TABLE:

    Block [line 12]
    SYM TABLE:
     a : int

